### PR TITLE
Add TR_ASSERT_FATAL to `generateS390MemoryReference`

### PR DESCRIPTION
**Pull request by simonameng**
Adding TR_ASSERT_FATAL to `generateS390MemoryReference` which use `MemoryReference` constructor directly and pass in `Node` type variable. So that it will be noticed when zero symbol reference node is passed in, which may lead to crash in `calcDisplacement` function.

Signed-off-by: simonameng <simonameng97@gmail.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
@genie-omr build zlinux,zos

File: compiler/z/codegen/OMRMemoryReference.cpp [hunk 1]
```diff
@@ -3474,12 +3473,14 @@ reuseS390MemoryReference(TR::MemoryReference *baseMR, int32_t offset, TR::Node *
 TR::MemoryReference *
 generateS390MemoryReference(TR::Node * node, TR::SymbolReference * sr, TR::CodeGenerator * cg)
    {
+   TR_ASSERT_FATAL(node->getOpCode().hasSymbolReference(), "Memory reference generation API needs a node with symbol reference\n");
```
Comment (NigelYiboYu) [inline compiler/z/codegen/OMRMemoryReference.cpp]: This API has a SymbolReference passed in and does not use the `node`'s symbolRef. Hence, this is not a memory reference without symbol reference. Hence, this line shouldn't be added.

File: compiler/z/codegen/OMRMemoryReference.cpp [hunk 2]
```diff
@@ -3474,12 +3473,14 @@ reuseS390MemoryReference(TR::MemoryReference *baseMR, int32_t offset, TR::Node *
...
    return new (cg->trHeapMemory()) TR::MemoryReference(node, sr, cg);
    }
 
 TR::MemoryReference *
 generateS390MemoryReference(TR::Snippet * tas, TR::CodeGenerator * cg, TR::Register * base, TR::Node * node)
    {
+   TR_ASSERT_FATAL(node->getOpCode().hasSymbolReference(), "Memory reference generation API needs a node with symbol reference\n");
    return new (cg->trHeapMemory()) TR::MemoryReference(tas, cg, base, node);
```
Comment (NigelYiboYu) [inline compiler/z/codegen/OMRMemoryReference.cpp]: This API uses ` new (cg->trHeapMemory()) TR::MemoryReference(tas, cg, base, node);`. If you go all the way down to this constructor, you'll see how it gets its symbol reference and a `TR_ASSERT_FATAL` is not needed here, either.

**Comment by fjeremic** [comment]
@genie-omr build zlinux,zos

**Pull request merged**